### DISCUSSION TITLE
Add parameter support for ipxe-x86_64.efi

### DISCUSSION THREAD

**Discussion by ccy** [ipxe/ipxe#721] [Ideas]
In built-in UEFI shell, EFI application allow passing parameters for execution.

If the `ipxe-x86_64.efi` accept command line parameters, perhaps it open more options like:

1. Allow to specify which network interface to boot with dhcp
2. Allow specify an initial ipxe boot script in addition to `autoexec.ipxe`.

**Comment by NiKiZe** [comment]
Ipxe.lkrn supports script on cmdline

That part is not yet ported to efi

**Comment by stappersg** [comment]
Seen the idea.  And I do see possible use cases.

@ccy what is your  intension with this idea, #721?
That question reworded:  @ccy what do you expect from reporting the idea here #721?

**Reply by ccy** [reply]
When I use the `ipxe-x86_64.efi`, I encounter an inconvenient situations.

I want to load an `ipxe` script in adhoc manner when launch the `ipxe-x86_64.efi` application in UEFI built-in shell.

**Reply by ccy** [reply]
One more use case here.

I have a machine equipped with 5 network interfaces: `net0`, `net1`, `net2`, `net3`, `net4`.  Only `net4` is connected with DHCP server.

When I run `ipxe-x86_64.efi` from UEFI shell, the `ipxe-x86_64.efi` application will try all network interface one by one start from `net0`.  It takes some time to wait for it to try `net4` interface.

If I already know the network interface, I can do something like this: `ipxe-x86_64.efi net4`

**Reply by readncode** [reply]
I ran into this limitation as well now newer PC;s  do no longer have a BIOS and can only use the  EFI.
It seems to me that [this patch]( https://lists.ipxe.org/pipermail/ipxe-devel/2017-September/005820.html) is just what is needed,, The thread does not make clear why it was never accepted.

**Reply by NiKiZe** [reply]
> [this patch](https://lists.ipxe.org/pipermail/ipxe-devel/2017-September/005820.html) is just what is needed,, The thread does not make clear why it was never accepted.

Because it is a "hack" more things to consider before being production ready. Follow the thread and you find a different patch, also with issues.

**Comment by stappersg** [comment]
2023-01-15:
> I ran into this limitation as well now newer PC;s do no longer have a BIOS and can only use the EFI.
It seems to me that [this patch](https://lists.ipxe.org/pipermail/ipxe-devel/2017-September/005820.html) is just what is needed.


That reminds on https://github.com/ipxe/ipxe/pull/631#issuecomment-1092864939

> The thread does not make clear why it was never accepted.

That was probably transmitted as

> The mailinglist thread does not make clear why it is **not yet** accepted.

**Comment by mcb30** [comment]
This has now been implemented in https://github.com/ipxe/ipxe/pull/871
